tests/test_imports.py

```diff
@@ -1,78 +1,81 @@
-import mne_connectivity
 
-import mne_faster
-import mne_nirs
-import mne_qt_browser
+def _import(mod):
+    try:
+        return importlib.import_module(mod.replace("-", "_"))
+    except Exception:
+        raise ImportError(f"Could not import {repr(mod)}")
 
-check_min_version(mne_qt_browser, "0.5.0")
-import mne_features
-import mne_rsa
-import mne_microstates
-import mne_ari
-import mne_kit_gui
-import mne_lsl
-import mne_icalabel
-import autoreject
 
-# import wfdb  # TODO
-import meegkit
-import eeg_positions
-import pyriemann
-import pyprep
-import pycrostates
-import darkdetect
-import qdarkstyle
-import numba
-import openpyxl
-import xlrd
-import pingouin
-import pactools
-import tensorpac
-import emd
-import neurodsp
-
-# import bycycle  # TODO
-import fooof
-import openneuro
-
-# import sleepecg  # TODO
-# import yasa  # TODO
-import neo
-import neurokit2
-import questionary
-import matplotlib.pyplot
-import seaborn
-import plotly
-import pqdm
-import pyvistaqt
-import vtk
-import PySide6.QtCore
-
-import matplotlib.pyplot as plt
+def check_version_eq(package, ver):
+    """Check a minimum version."""
+    from packaging.version import parse
 
-backend = plt.get_backend()
-assert backend.lower() == "qtagg", backend
+    if isinstance(package, str):
+        package = _import(package)
+
+    try:
+        package_ver = parse(package.__version__)
+    except Exception:
+        raise ImportError(
+            f"Could not parse version for {package}: {repr(package.__version__)}"
+        )
+    assert package_ver >= parse(ver), (
+        f"{package}: got {package.__version__} wanted {ver}"
+    )
 
-check_min_version(pyvistaqt, "0.11.0")
 
+# All related software
+lines = (
+    (Path(__file__).parents[1] / "recipes" / "mne-python" / "construct.yaml")
+    .read_text("utf-8")
+    .splitlines()
+)
+lines = [line.strip() for line in lines]
+all_lines = lines
+sidx = lines.index("# <<< BEGIN RELATED SOFTWARE LIST >>>")
+eidx = lines.index("# <<< END RELATED SOFTWARE LIST >>>")
+lines = [line for line in lines[sidx : eidx + 1] if not line.startswith("#")]
+assert lines
+mods = [line[2:].split("#")[0].split(">")[0].split("=")[0].strip() for line in lines]
+
+# Plus some custom ones
+mods += """
+darkdetect qdarkstyle numba openpyxl xlrd pingouin questionary
+seaborn plotly pqdm pyvistaqt vtk PySide6.QtCore matplotlib.pyplot
+""".strip().split()
 if platform.system() == "Darwin":
-    import Foundation  # pyobjc
+    mods += ["Foundation"]  # pyobjc
+
+# Now do the importing and version checking
+bad_ver = {
+    "mne-faster",  # https://github.com/wmvanvliet/mne-faster/pull/7
+    "mne-ari",  # https://github.com/john-veillette/mne-ari/pull/7
+}
+for mod in tqdm(mods, desc="Imports", unit="module"):
+    if mod in parsed.ignore:
+        continue
+    use_mod = mod
+    if use_mod.startswith("python-"):  # python-neo
+        use_mod = use_mod[7:]
+    if use_mod.endswith("-py"):  # openneuro-py
+        use_mod = use_mod[:-3]
```

Comment:
if this grows beyond 2 items, I'd switch to using a lookup dict:

```py
mod_name_changes = {"python-neo": "neo", "openneuro-py": "openneuro"}
use_mod = mod_name_changes.get(use_mod, use_mod)
```

Reply:
Yeah it was 3, two of which were `python-` so I figured it was better than a dict